src/main/java/org/gridsuite/shortcircuit/server/repositories/specifications/SpecificationUtils.java

```diff
@@ -45,18 +45,33 @@ public static <X> Specification<X> startsWith(String field, String value) {
         return (root, cq, cb) -> cb.like(cb.upper(getColumnPath(root, field)), EscapeCharacter.DEFAULT.escape(value).toUpperCase() + "%", EscapeCharacter.DEFAULT.getEscapeCharacter());
     }
 
-    public static <X> Specification<X> notEqual(String field, Double value) {
+    public static <X> Specification<X> notEqual(String field, Integer value) {
         return (root, cq, cb) -> cb.notEqual(getColumnPath(root, field), value);
     }
 
-    public static <X> Specification<X> lessThanOrEqual(String field, Double value) {
+    public static <X> Specification<X> notEqual(String field, Double value, Double tolerance) {
+        return (root, cq, cb) -> cb.or(
+                cb.greaterThan(getColumnPath(root, field), value + tolerance),
+                cb.lessThan(getColumnPath(root, field), value - tolerance)
+        );
+    }
+
+    public static <X> Specification<X> lessThanOrEqual(String field, Integer value) {
         return (root, cq, cb) -> cb.lessThanOrEqualTo(getColumnPath(root, field), value);
     }
 
-    public static <X> Specification<X> greaterThanOrEqual(String field, Double value) {
+    public static <X> Specification<X> lessThanOrEqual(String field, Double value, Double tolerance) {
+        return (root, cq, cb) -> cb.lessThanOrEqualTo(getColumnPath(root, field), value + tolerance);
+    }
+
+    public static <X> Specification<X> greaterThanOrEqual(String field, Integer value) {
         return (root, cq, cb) -> cb.greaterThanOrEqualTo(getColumnPath(root, field), value);
     }
 
+    public static <X> Specification<X> greaterThanOrEqual(String field, Double value, Double tolerance) {
+        return (root, cq, cb) -> cb.greaterThanOrEqualTo(getColumnPath(root, field), value - tolerance);
+    }
+
```

Comment:
I'm a bit confused with including tolerance without documenting it or renaming the methods
I don't what would be the best though, maybe a bit comment above all methods, of above each of them

Reply:
with the variable name is normally self-explanatory, but I add a comment to make it even clearer

Reply:
ths thing is we have twice the method greaterThanOrEqual, and they don't work the same way
I think "tolerance" behaviour should not be hidden